### PR TITLE
Specify tempfile random naming constraints

### PR DESCRIPTION
**Pull request by alanyee**
Specify which random characters are used during temporary file name creation. Trivial PR, but it may be helpful for developers to know what characters are used during file creation. Source: [`class _RandomNameSequence`](https://github.com/python/cpython/blob/master/Lib/tempfile.py#L128)

### CODE REVIEW

**Comment by alanyee** [comment]
I would like this added to the documentation.

**Comment by alanyee** [comment]
@vstinner may I ask you to review this PR?

File: Doc/library/tempfile.rst
Comment (slateny) [inline Doc/library/tempfile.rst]: For Sphinx, the grave signs needed to be doubled up (` ``a-z`` `)

**Review by slateny — APPROVED** [review]
Lgtm, though I wonder if this should be spelled out (lowercase alphanumeric characters plus the underscore) instead of using a regex-like expression, since I also couldn't find many examples of `a-z` being used outside of a regex context. Not too sure which is better though, so I'll let someone else chime in on this

**Comment by hauntsaninja** [comment]
Thanks for the PR, but this feels like an implementation detail. Would you mind talking more about the use case you have for the exact set of characters used to be a documented guarantee?

**Comment by alanyee** [comment]
@hauntsaninja I don't exactly want a guarantee or whatnot. At the time of creating this PR, I want to know what the tempfile's randomly generated names would look like, and for my use case at the time, I wanted to generate my own files ensuring I would confuse them for the tempfile's random names. So I decided to look at the source code to see what kind of characters would be used to create the tempfiles, so I decided to document it so that others who had the same inquiry as me would know.

**Comment by hauntsaninja** [comment]
Okay, thanks, that helps me better understand the use case. I think the right way to disambiguate temporary files from other files is to use the `prefix` and `suffix` arguments. I'd much rather encourage users to use those arguments than e.g. rely on the fact that the implementation doesn't currently use a certain character in filenames.

Putting statements in the documentation is essentially a guarantee. I appreciate your intention to save time for others, but curiosity about implementation details is still best served by looking at the implementation (one of my favourite things about Python is that the standard library is quite readable).

**Pull request closed without merging**
